### PR TITLE
[scarb-doc] Add initial crate structures

### PR DESCRIPTION
**Pull request by drknzz**
Next PR will add nested attributes to the structures:
- struct members
- enum variants
- impl items
- trait items

### CODE REVIEW

File: extensions/scarb-doc/src/types.rs [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+            full_path: id.full_path(db),
+            doc: db.get_item_documentation(LookupItemId::ModuleItem(id)),
+            definition: db.get_item_definition(LookupItemId::ModuleItem(id)),
+            text: node.as_syntax_node().get_text_without_trivia(db.upcast()),
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Constant {
```
Comment (Arcticae) [inline extensions/scarb-doc/src/types.rs:159]: I feel like all of those defs + impls are repetitive and could be done with a macro with like 3 input vars (pointer type, module type and the ModuleItemId)

File: extensions/scarb-doc/src/types.rs
Comment (Arcticae) [inline extensions/scarb-doc/src/types.rs:1]: [nitpick]: Those types are really nice, but i would try to give them each a docstring, for an example code snippet that they represent.

File: extensions/scarb-doc/src/main.rs [hunk 2]
```diff
@@ -42,166 +45,38 @@ fn main() -> Result<()> {
...
-    let crates_config = get_crates_config(metadata, compilation_unit_metadata);
-
-    ProjectConfig {
-        base_path: package_metadata.root.clone().into(),
-        corelib: Some(Directory::Real(corelib.source_root().into())),
-        content: ProjectConfigContent {
-            crate_roots: dependencies,
-            crates_config,
-        },
+fn print_module(module: &types::Module) {
```
[the remark this answers is not in the document]
Comment (Arcticae) [reply extensions/scarb-doc/src/main.rs:56]: I also feel like sub-structs could also have `Display`s and so on...

File: extensions/scarb-doc/src/compilation.rs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+use cairo_lang_compiler::project::{AllCratesConfig, ProjectConfig, ProjectConfigContent};
+use cairo_lang_filesystem::cfg::{Cfg, CfgSet};
+use cairo_lang_filesystem::db::{CrateSettings, Edition, ExperimentalFeaturesConfig};
+use cairo_lang_filesystem::ids::Directory;
+use cairo_lang_utils::ordered_hash_map::OrderedHashMap;
+
+const LIB_TARGET_KIND: &str = "lib";
+const CORELIB_CRATE_NAME: &str = "core";
+
+pub fn get_project_config(
```
[the remark this answers is not in the document]
Comment (drknzz) [reply extensions/scarb-doc/src/compilation.rs:16]: Mainly yes, some slight refactor changes

File: extensions/scarb-doc/src/types.rs [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+
+#[derive(Clone, Debug)]
+pub struct ItemData {
+    pub name: String,
+    pub full_path: String,
+
+    pub doc: Option<String>,
+    pub definition: String,
+
+    pub text: String,
```
[the remark this answers is not in the document]
Comment (drknzz) [reply extensions/scarb-doc/src/types.rs:143]: It differs for item types but for example, a free function's `definition` is its signature while `text` is the whole function

File: extensions/scarb-doc/src/types.rs (hunk 1 above)
Comment (drknzz) [reply extensions/scarb-doc/src/types.rs:159]: ikr, but they are planned to change and have some separate attributes, so I kept them like this for now

File: extensions/scarb-doc/src/main.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (drknzz) [reply extensions/scarb-doc/src/main.rs:56]: True, but there's a good chance it'll be removed soon anyway so I didn't bother D; (as you've mentioned it's only for testing)

**Review by Arcticae — APPROVED** [review]

**Pull request merged**